docs/release_process.rst

```diff
@@ -26,30 +26,29 @@ As such, patch releases should only ever be based on an existing release point.
    in the ``Z`` version component.
 
    a. Use ``poetry version patch`` to increase the patch value appropriately in
-      the :file:`pyproject.toml` file.
+      the :file:`pyproject.toml` file. This is important because the automated
+      release process will try to publish the version listed in this file and it
+      will error out if the version it tries to upload to PYPI already exists.
 
    b. Rename the ``docs/release_notes/pending_patch.rst`` file to
       ``docs/release_notes/v{VERSION}.rst``, matching the new version value.
       Add a descriptive paragraph under the title section summarizing this
-      release.
+      release. During the automated process, this version file will referenced
+      to create the header of the release on Github.
 
    c. Add a reference to the new release notes RST file in
       ``docs/release_notes.rst``.
 
    d. In a separate commit, add back a blank pending release notes file stub.
       See `Stub Pending Notes File`_.
 
-2. Create a pull/merge request for this branch with master as the merge target.
-   This is to ensure that everything passes CI testing before making the
-   release. If there is an issue then branches should be made and merged into
-   this branch until the issue is resolved.
+2. Merge version bump branch into ``master`` branch.
 
-3. Tag branch (see `Tag new version`_ below ) after resolving issues and before
-   merging into ``master``.
-
-4. Merge version bump branch into ``master`` branch.
-
-5. `Create new version release to PYPI`_
+3. Tag branch from the command line (see `Tag new version`_ below ). After the
+   tag is pushed, a CI check and build will automatically run. Should everything
+   pass without issue, the build will be published on PYPI. If the build or
+   tests fail, then an issue and branch should be created to address and solve
+   the errors before proceeding with the publishing of the branch.
```

Comment:
Should this instead be *before* merging? (1) errors during release can be corrected while still on a branch instead of introducing an error point on the master line. (2) We want to probably be clear/sure that tagging occurs on the last commit of of this "bump" branch, and *not* on the merge-commit into master: master may have moved since the creation of the release branch and rebasing/updating the release branch may not applicable / make sense (there are possible ramifications to the current merge-policy, but that's maybe a different topic).